[Bug] "com.mucommander.ui.main.FolderPanel.getLocationTextField()" is null

### Is there an existing issue for this?

- [x] I have searched the existing issues (including the closed ones)

### Description

At the start of the `muCommander`, **only** the splash screen is visible.
And no more.
After completing the process via kill and restarting, all previous tabs were **deleted** :(

### Steps to reproduce

Update to version 1.6.1, and start `muCommander` after restart computer.
(_That's how it showed up for me._)

### Expected behavior

The panels were expected to load and display as usual.

### Actual behavior

The screen saver is shown and then nothing.

### Screenshots?

**Only** the splash screen and no more.

### muCommander version

Version: 1.6.1
Build date: 2026/04/21
Build number: 1
Build metadata: 7057ad9

### Java version

Runtime version: 25.0.2
VM name: OpenJDK 64-Bit Server VM
VM version: 25.0.2+10-LTS
VM vendor: Eclipse Adoptium

### Operating System type and version

Name: Linux Debian 13.3
Version: 6.12.74+deb13+1-amd64
Architecture: amd64

### Relevant log output

```shell
Preloading fonts completed in: 2198ms
[muEpoch+2430ms] muCommander: OSGi framework with muC fully started
Exception in thread "MainFrameInit" java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "com.mucommander.ui.main.LocationTextField.r
equestFocus()" because the return value of "com.mucommander.ui.main.FolderPanel.getLocationTextField()" is null
        at com.mucommander.ui.main.MainFrame.<init>(MainFrame.java:332)
        at com.mucommander.ui.main.frame.DefaultMainFramesBuilder.createMainFrame(DefaultMainFramesBuilder.java:121)
        at java.base/java.util.stream.IntPipeline$1$1.accept(Unknown Source)
        at java.base/java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.Spliterator$OfInt.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
        at com.mucommander.ui.main.frame.DefaultMainFramesBuilder.build(DefaultMainFramesBuilder.java:67)
        at com.mucommander.ui.main.WindowManager.createNewMainFrame(WindowManager.java:205)
        at com.mucommander.Application.lambda$run$12(Application.java:496)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "com.mucommander.ui.main.LocationTextField.requestFocus()" because the return value of "com.mucom
mander.ui.main.FolderPanel.getLocationTextField()" is null
        at java.base/java.util.concurrent.FutureTask.report(Unknown Source)
        at java.base/java.util.concurrent.FutureTask.get(Unknown Source)
        at com.mucommander.ui.main.MainFrame.init(MainFrame.java:191)
        at com.mucommander.ui.main.MainFrame.<init>(MainFrame.java:330)
        ... 13 more
Caused by: java.lang.NullPointerException: Cannot invoke "com.mucommander.ui.main.LocationTextField.requestFocus()" because the return value of "com.mucommander.ui.main.FolderPanel.getLocationTex
tField()" is null
        at com.mucommander.ui.main.tabs.FileTableTabs.setTabsViewer(FileTableTabs.java:68)
        at com.mucommander.ui.tabs.HideableTabbedPane.switchToTabsWithHeaders(HideableTabbedPane.java:246)
        at com.mucommander.ui.tabs.HideableTabbedPane.refreshViewer(HideableTabbedPane.java:290)
        at com.mucommander.ui.tabs.HideableTabbedPane.tabAdded(HideableTabbedPane.java:312)
        at com.mucommander.ui.tabs.TabsCollection.fireTabAdded(TabsCollection.java:172)
        at com.mucommander.ui.tabs.TabsCollection.add(TabsCollection.java:83)
        at com.mucommander.ui.tabs.TabsCollection.add(TabsCollection.java:72)
        at com.mucommander.ui.tabs.HideableTabbedPane.addTab(HideableTabbedPane.java:171)
        at com.mucommander.ui.main.tabs.FileTableTabs.lambda$new$0(FileTableTabs.java:60)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.forEach(Unknown Source)
        at com.mucommander.ui.main.tabs.FileTableTabs.<init>(FileTableTabs.java:60)
        at com.mucommander.ui.main.FolderPanel.<init>(FolderPanel.java:194)
        at com.mucommander.ui.main.MainFrame.lambda$new$5(MainFrame.java:329)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        ... 1 more
```

--- a/mucommander-core/src/main/java/com/mucommander/ui/main/tabs/FileTableTabs.java
+++ b/mucommander-core/src/main/java/com/mucommander/ui/main/tabs/FileTableTabs.java
@@ -17,8 +17,6 @@
 
 package com.mucommander.ui.main.tabs;
 
-import java.util.Arrays;
-
 import com.mucommander.commons.file.AbstractFile;
 import com.mucommander.commons.file.FileURL;
 import com.mucommander.ui.event.LocationEvent;
@@ -29,6 +27,8 @@ import com.mucommander.ui.tabs.HideableTabbedPane;
 import com.mucommander.ui.tabs.TabFactory;
 import com.mucommander.ui.tabs.TabsViewerFactory;
 
+import java.util.Arrays;
+
 /**
 * HideableTabbedPane of {@link com.mucommander.ui.main.tabs.FileTableTab} instances.
 * 
@@ -37,7 +37,7 @@ import com.mucommander.ui.tabs.TabsViewerFactory;
 public class FileTableTabs extends HideableTabbedPane<FileTableTab> implements LocationListener {
 
 	/** FolderPanel containing those tabs */
-	private FolderPanel folderPanel;
+	private final FolderPanel folderPanel;
 
 	/** Factory of instances of FileTableTab */
 	private DefaultFileTableTabFactory defaultTabsFactory;
@@ -62,10 +62,14 @@ public class FileTableTabs extends HideableTabbedPane<FileTableTab> implements L
 
 	@Override
 	protected void setTabsViewer(TabsViewerFactory<FileTableTab> tabsViewerFactory) {
-		// During the transition to the other viewer, the focus might switch to the
-		// previously focused component, which may be in the other panel. Hence we switch
-		// the focus to the location text field so that it remains at the current panel.
-		folderPanel.getLocationTextField().requestFocus();
+		var locationTextField = folderPanel.getLocationTextField();
+		// locationTextField might be null during initialization (created in background thread)
+		if (locationTextField != null) {
+			// During the transition to the other viewer, the focus might switch to the
+			// previously focused component, which may be in the other panel. Hence we switch
+			// the focus to the location text field so that it remains at the current panel.
+			locationTextField.requestFocus();
+		}
 		super.setTabsViewer(tabsViewerFactory);
 	}
 

--- a/package/readme.txt
+++ b/package/readme.txt
@@ -44,7 +44,7 @@ Localization:
 -
 
 Bug fixes:
--
+- Fixed a possible NPE during initialization that prevents the application from starting.
 
 Known issues:
 - Some translations may not be up-to-date.

